### PR TITLE
Use `num_streams` as parallelism when read thread is enabled.

### LINKED ISSUE #6343: We can use `num_streams` as parallelism when read thread is enabled.
## Enhancement

https://github.com/pingcap/tiflash/blob/7203151a59ac7c44c673fc5f689fef312c660731/dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp#L904

https://github.com/pingcap/tiflash/blob/7203151a59ac7c44c673fc5f689fef312c660731/dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp#L922-L951

The `num_streams` is passed down from the compute layer, and the value is usually physical cpu core num.
We can use `num_streams` directly when read thread is enabled.

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: close #6343

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by gengliqi — APPROVED** [review]

File: dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp [hunk 1]
```diff
@@ -919,21 +920,23 @@ BlockInputStreams DeltaMergeStore::readRaw(const Context & db_context,
...
+                req_info));
         }
-        else
+    }
+    else
+    {
+        for (size_t i = 0; i < final_num_stream; ++i)
         {
-            stream = std::make_shared<DMSegmentThreadInputStream>(
+            res.push_back(std::make_shared<DMSegmentThreadInputStream>(
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp:939]:
```suggestion
            res.emplace_back(std::make_shared<DMSegmentThreadInputStream>(
```

**Review by JaySon-Huang — APPROVED** [review]
LGTM with a minor comment

**Pull request merged**